apps/wrf/readme.md

```diff
@@ -1,110 +1,461 @@
+./ungrib.exe >& ungrib.log
 
-cd to your WPS installation directory and copy the new_conus2.5km namelist.wps to this location.
+#### Run metgrid.exe
+mpirun --allow-run-as-root ./metgrid.exe
+```
 
-Download WPS v4 geopraphical static data for WPS v4. (https://www2.mmm.ucar.edu/wrf/users/download/get_sources_wps_geog.html), Download topo_gmted2010_30s, modis_landuse_20class_30s_with_lakes, soiltemp_1deg, soiltype_top_30s, albedo_modis, greenfrac_fpar_modis, lai_modis_10m, maxsnowalb_modis_10m and orogwd_10m. Extract into a directory.
+Expected results 1: 
+``` 
+# !  Successful completion of metgrid.  ! 
+# You may receive also some warnings: 
+Note: The following floating-point exceptions are signalling: IEEE_OVERFLOW_FLAG IEEE_UNDERFLOW_FLAG IEEE_DENORMAL
+``` 
+Expected results 2: 
+You will find the files below:
+``` 
+ls -l *met_em.d*
+# Should see the following files:
+# met_em.d01.2018-06-17_12:00:00.nc
+# met_em.d01.2018-06-17_09:00:00.nc
+# met_em.d01.2018-06-17_06:00:00.nc
+# met_em.d01.2018-06-17_03:00:00.nc
+# met_em.d01.2018-06-17_00:00:00.nc
 ```
-mpirun ./geogrid.exe
+
+#### Run real.exe
+``` 
+# Keep as root
+sudo su -
+
+cd /apps/hbv2/wrf-openmpi/WRF-4.1.5/run
+cp -f namelist.input namelist.input.old
+cp -f /data/wrfdata/WRF_benchmarks/cases/new_conus2.5km/namelist.input .
+cp /apps/hbv2/wps-openmpi/WPS-4.1/met_em.d0*.nc .
+mpirun --allow-run-as-root ./real.exe
 ```
-The file geo_em.d01.nc should be created.
+
+Expected results: 
+The following files should be generated in /apps/hbv2/wrf-openmpi/WRF-4.1.5/run/
+- wrfbdy_d01
+- wrfinput_d01
 ```
-ln -s ungrib/Variable_Tables/Vtable.GFS Vtable
+ls -l /apps/hbv2/wrf-openmpi/WRF-4.1.5/run/*_d0*
 ```
+
+#### Change Permissions on Files
+Grant permission to users that will run the WRF to access the files on /data and /apps. 
+For my environment, I used "azureadmin" as CycleCloud admin. 
+Change the user below accordingly.     
+
 ```
-./link_grib.csh /location/of/NCEP_GFS_Model_Run_data/gfs.0p25.20180617*
+# grant permission to users that will run the WRF to access the files. 
+chown -R <your admin user>:cyclecloud /data
+chown -R <your admin user>:cyclecloud /apps
+chmod -R g+w /data
+chmod -R g+w /apps
 ```
+
+**If you get here, you have completed the WRF v4 setup!**
+
+Now you can shutdown and terminate the HBv2 worker node used to perform these setup procedures.
+
+## Running and Testing
+
+Connect to head node of WRF Cluster and submit WRF v4 simulation job:
+
+> Where SKU_TYPE is the sku type you are running on and INPUTDIR contains the location of wrf input files (namelist.input, wrfbdy_d01 and wrfinput_d01)
+
+- Test 1 
+  - SKU: Standard_HB120rs_v2
+  - Nodes: 1 
+  - Processes per node: 60 
+  - MPI processes per node: 60 
 ```
-./ungrib.exe >& ungrib.log
+mkdir ~/test1
+cd ~/test1
+
+qsub -l select=1:nodearray=execute1:ncpus=60:mpiprocs=60,place=scatter:excl -v "SKU_TYPE=hbv2,INPUTDIR=/apps/hbv2/wrf-openmpi/WRF-4.1.5/run" /data/azurehpc/apps/wrf/run_wrf_openmpi.pbs
```

Comment:
For HBv2, should ncpus=120 ?

Reply:
These were tests I did to measure the execution time with different configs. I forgot to add the execution time duration results. I will add a chart with it.